column.go

```diff
@@ -486,97 +487,114 @@ func schemaRepetitionTypeOf(s *format.SchemaElement) format.FieldRepetitionType
 	return format.Required
 }
 
-func (c *Column) decompress(data []byte, size int32) ([]byte, error) {
-	var page []byte
-	if size > 0 {
-		page = make([]byte, size)
+func (c *Column) decompress(compressedPageData []byte, compressedPageSize int32) (page *buffer, err error) {
+	page = uncompressedPageBufferPool.get()
+	if compressedPageSize > 0 {
+		page.resize(int(compressedPageSize))
```

Comment:
Semantically, confusing why we'd take a buffer from the uncompressed page buffer pool, but set its size to the compressed size - seems like we'd want to set the size to the uncompressed size instead.

Reply:
Thanks for pointing it out, it's confusing because it's wrong, I should have named the variable `uncompressedPageSize` 👍